src/zeta/aca_oam_port_manager.cpp

```diff
@@ -99,60 +99,56 @@ void Aca_Oam_Port_Manager::_clear_all_data()
 void Aca_Oam_Port_Manager::create_entry_unsafe(uint port_number)
 {
   ACA_LOG_DEBUG("%s", "Aca_Oam_Port_Manager::create_entry_unsafe ---> Entering\n");
-
-  unordered_set<uint> tunnel_ids_table;
-  _oam_ports_table.emplace(port_number, tunnel_ids_table);
+  
+  _oam_ports_cache.emplace(port_number);
 
   ACA_LOG_DEBUG("%s", "Aca_Oam_Port_Manager::create_entry_unsafe <--- Exiting\n");
 }
 
 // update oam_ports_table and add the OAM punt rule also if this is the first port in the VPC
-void Aca_Oam_Port_Manager::add_vpc(uint port_number, uint tunnel_id)
+void Aca_Oam_Port_Manager::add_port_number(uint port_number)
 {
-  ACA_LOG_DEBUG("%s", "Aca_Oam_Port_Manager::add_vpc ---> Entering\n");
+  ACA_LOG_DEBUG("%s", "Aca_Oam_Port_Manager::add_port_number ---> Entering\n");
   // -----critical section starts-----
-  _oam_ports_table_mutex.lock();
-  if (_oam_ports_table.find(port_number) == _oam_ports_table.end()) {
+  _oam_ports_cache_mutex.lock();
+  if (_oam_ports_cache.find(port_number) == _oam_ports_cache.end()) {
     create_entry_unsafe(port_number);
     _create_oam_ofp(port_number);
   }
-  _oam_ports_table[port_number].emplace(tunnel_id);
+  _oam_ports_cache.emplace(port_number);
 
-  _oam_ports_table_mutex.unlock();
+  _oam_ports_cache_mutex.unlock();
   // -----critical section ends-----
 
-  ACA_LOG_DEBUG("%s", "ACA_OVS_Programmer::add_vpc <--- Exiting\n");
+  ACA_LOG_DEBUG("%s", "ACA_OVS_Programmer::add_port_number <--- Exiting\n");
 }
 
 // update oam_ports_table and delete the OAM punt rule if the last port in the VPC has been deleted
-int Aca_Oam_Port_Manager::remove_vpc(uint port_number, uint tunnel_id)
+int Aca_Oam_Port_Manager::remove_port_number(uint port_number)
 {
-  ACA_LOG_DEBUG("%s", "Aca_Oam_Port_Manager::remove_vpc ---> Entering\n");
+  ACA_LOG_DEBUG("%s", "Aca_Oam_Port_Manager::remove_port_number ---> Entering\n");
 
   int overall_rc;
 
   // -----critical section starts-----
-  _oam_ports_table_mutex.lock();
-  if (_oam_ports_table.find(port_number) == _oam_ports_table.end()) {
+  _oam_ports_cache_mutex.lock();
+  if (_oam_ports_cache.find(port_number) == _oam_ports_cache.end()) {
     ACA_LOG_ERROR("port id %u not find in oam_ports_table\n", port_number);
     overall_rc = ENOENT;
   } else {
-    _oam_ports_table[port_number].erase(tunnel_id);
     // clean up the oam_ports_table entry and oam flow rule if there is no port assoicated
-    if (_oam_ports_table[port_number].empty()) {
-      if (_oam_ports_table.erase(port_number) == 1 && _delete_oam_ofp(port_number) == EXIT_SUCCESS) {
-        ACA_LOG_INFO("Successfuly cleaned up entry for port_number %u\n", port_number);
-        overall_rc = EXIT_SUCCESS;
-      } else {
-        ACA_LOG_ERROR("Failed to clean up entry for port_number %u\n", port_number);
-        overall_rc = EXIT_FAILURE;
-      }
+    if (_oam_ports_cache.erase(port_number) == 1 && _delete_oam_ofp(port_number) == EXIT_SUCCESS) {
```

Comment:
line 139: fix comment to remove "if there is no port assoicated"

line 140: let's do the two operations in to seperate steps to avoid confusions.
_oam_ports_cache.erase(port_number)
_delete_oam_ofp(port_number)